### PR TITLE
Update criteo_dataset.py

### PR DESCRIPTION
**Pull request by soma2000-lang**
## Relevant Issue
#165

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/NimbleEdge/EnvisEdge/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/NimbleEdge/EnvisEdge/blob/main/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [NimbleEdge Styleguide](https://numpydoc.readthedocs.io/en/latest/format.html)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/NimbleEdge/EnvisEdge/labels)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by soma2000-lang** [comment]
@divyanshugit  Please look into it.

File: datasets/criteo/criteo_dataset.py [hunk 1]
```diff
@@ -4,20 +4,49 @@
 import numpy as np
 # pytorch
 import torch
-from torch.utils.data import Dataset
-
-
+from torch.utils.data import Dataset 
 class CriteoDataset(Dataset):
-    def __init__(
-            self,
```
Comment (divyanshugit) [inline datasets/criteo/criteo_dataset.py:12]: Instead of removing this from here. You can simply write the comments before this line.

File: datasets/criteo/criteo_dataset.py [hunk 2]
```diff
@@ -4,20 +4,49 @@
...
+ """
+    It's a simple dataloader for criteo
```
Comment (divyanshugit) [inline datasets/criteo/criteo_dataset.py:10]: It's a custom dataset for criteo. Please go through [this](https://pytorch.org/tutorials/beginner/basics/data_tutorial.html). So, you'll have a better understanding about this block of code.

File: datasets/criteo/criteo_dataset.py [hunk 3]
```diff
@@ -4,20 +4,49 @@
...
+    collate_wrapper_criteo_offset
+    offset_to_length_converter  Changes from offset to length 
+    make_criteo_data_and_loader 
+    """
+
+    def __init__(                       
+            self,                   
             X_int, X_cat, y,
-            max_ind_range):
+            max_ind_range):
```
Comment (divyanshugit) [inline datasets/criteo/criteo_dataset.py:31]: Why?

File: datasets/criteo/criteo_dataset.py [hunk 4]
```diff
@@ -4,20 +4,49 @@
...
+    """
+
+    def __init__(                       
+            self,                   
             X_int, X_cat, y,
-            max_ind_range):
+            max_ind_range):               
         self.max_ind_range = max_ind_range
-        self.X_int = X_int
+        self.X_int = X_int  #initializing or defining the variables
```
Comment (divyanshugit) [inline datasets/criteo/criteo_dataset.py:33]: We're already writing the inline docs for this. So, we don't have to define it in this way. [ditto - I]

File: datasets/criteo/criteo_dataset.py [hunk 5]
```diff
@@ -4,20 +4,49 @@
...
         self.X_cat = X_cat
         self.y = y
 
-    def __getitem__(self, index):
+    def __getitem__(self, index):
```
Comment (divyanshugit) [inline datasets/criteo/criteo_dataset.py:37]: What did you change in this line?

File: datasets/criteo/criteo_dataset.py [hunk 6]
```diff
@@ -26,7 +55,7 @@ def __getitem__(self, index):
                 )
             ]
 
-        if self.max_ind_range > 0:
+        if self.max_ind_range > 0: #range should be greater than zero for returning the followinng paarmeters
```
Comment (divyanshugit) [inline datasets/criteo/criteo_dataset.py:58]: ditto - I

File: datasets/criteo/criteo_dataset.py [hunk 7]
```diff
@@ -36,60 +65,117 @@ def __getitem__(self, index):
...
+        """
         X_int = torch.log(torch.tensor(X_int, dtype=torch.float) + 1)
         if self.max_ind_range > 0:
             X_cat = torch.tensor(X_cat % self.max_ind_range, dtype=torch.long)
         else:
             X_cat = torch.tensor(X_cat, dtype=torch.long)
         y = torch.tensor(y.astype(np.float32))
 
-        return X_int, X_cat, y
+        return X_int, X_cat, y
```
Comment (divyanshugit) [inline datasets/criteo/criteo_dataset.py:87]: Why are you adding a space at the end of this line?

File: datasets/criteo/criteo_dataset.py [hunk 8]
```diff
@@ -36,60 +65,117 @@ def __getitem__(self, index):
...
         y = torch.tensor(y.astype(np.float32))
 
-        return X_int, X_cat, y
+        return X_int, X_cat, y 
 
     def __len__(self):
-        return len(self.y)
+        """Returns the length"""
+      
+        return len(self.y)
```
Comment (divyanshugit) [inline datasets/criteo/criteo_dataset.py:92]: Unnecessary new lines. You can mention this while writing the purpose of the class.

**Review by divyanshugit — CHANGES_REQUESTED** [review]

**Comment by divyanshugit** [comment]
@soma2000-lang Instead of removing previous lines and adding lines. You can just add relevant things at relevant places.
